perun-core/src/main/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntry.java

```diff
@@ -398,6 +396,29 @@ public void copyOwners(PerunSession sess, Facility sourceFacility, Facility dest
 		return getFacilitiesManagerBl().getAssignedResources(sess, facility);
 	}
 
+	@Override
+	public List<Resource> getAssignedResourcesByAssignedService(PerunSession sess, Facility facility, Service service) throws PrivilegeException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getAssignedResourcesByAssignedService_Facility_Service_policy", facility, service)) {
+			throw new PrivilegeException(sess, "getAssignedResourcesByAssignedService");
+		}
+
+		List<Resource> allResources = getFacilitiesManagerBl().getAssignedResources(sess, facility);
+		List<Resource> resources = new ArrayList<>();
+		List<Service> services;
+
+		for (Resource resource: allResources) {
+			services = getPerunBl().getResourcesManagerBl().getAssignedServices(sess, resource);
+			if(services.contains(service)) {
+				resources.add(resource);
+			}
+		}
```

Comment:
I have noticed, there is a method that can be used for this purpose.
```
List<Resource> getAssignedResources(PerunSession perunSession, Facility facility, Vo specificVo, Service specificService);
```
Just put `null` to the specificVo parameter and it should work.

Reply:
Good point. I replaced the block of code by this method. I also added some checks for facility and service.